File: 01-search_in_two_array/search_in_two_array.php
<?php
// 第一种方法
function Find($target, $array)
{
    foreach($array as  $v){
        if(in_array($target,$v)){
            return 'true';
        }
    }
    return 'false';
}

while(fscanf(STDIN, "%d,%s", $a,$b) == 2){
    try{
        $target = $a;
        eval('$array = '.$b.";");
        echo Find($target, $array)."\n";
    }catch(Exception $e){
        die;
    }
}


// 第二种方法
function Find($target, $array){
    $i = count($array)-1;
    $j = 0;
    while($i>=0 && $j<count($array[0])){
        if($array[$i][$j] > $target)
            $i--;
        elseif($array[$i][$j] < $target)
            $j++;
        else
            return 1;
    }
    return 0;
}



?>

File: 01-search_in_two_array/search_in_two_array.java
public class Solution {
    public boolean Find(int target, int [][] array) {
        int len = array.length-1;   //列
        int i = 0;  //行
        while(len >= 0 && i < array[0].length){
            if(array[len][i] > target)
                len--;
            else if(array[len][i] < target)
                i++;
            else
               return true;
        }
        return false;
    }
}